docs/DESKTOP_ENVIRONMENTS.md

```diff
@@ -0,0 +1,118 @@
+   (the PID may be included optionally, simplifying the unit creation as it
+   can be done by setting `Slice=%N.slice` in a drop-in configuration).
+ * If no application ID is available, the launcher should generate a reasonable
+   name when possible (e.g. using `basename(argv[0])`).
+
+TODO: Define the name of slices that should be used. This could be `apps-launcher-ApplicationID.slice`.
+
+This has the following advantages:
+ * Using the `apps-launcher-` prefix means that the unit defaults can be adjusted
+   using desktop environment specific drop-in files.
+ * The application ID can be retrieved by stripping the prefix and postfix.
+   This in turn should map to the corresponding `.desktop` file when available
+
+## XDG autostart integration
+
+To allow XDG autostart integration, systemd will ship a cross-desktop generator
+to create appropriate units for the autostart directory. Desktop Environments
+will be able to make use of this simply by starting the appropriate XDG related
+targets. The names and ordering rules for these targets are to be defined.
+
+This generator will likely only support a subset of features. In particular,
+the GNOME specific feature to bind a service to a settings variable will not
+be supported.
+
+## Application managed cgroups
+
+Applications like browsers should place major components that are run as
+separate processes into their own cgroup. This can be useful to the
+applications themselves to apply further resource restrictions (e.g. disallow
+network access or memory limits). Furthermore, it exposes the structure of
+the application to the system, allowing it to e.g. kill entire cgroups in
+response to OOM situations without affecting the application as a whole.
+
+It is to be defined how these applications should create and manage these
+cgroups. Possible approaches could be:
+
+ * Directly use systemd and explicitly place items into a common slice. This
+   has the disadvantage of requiring privileged access to systemd.
+ * Provide an XDG portal, that allows creating relevant transient units. This
+   portal can enforce placement into the appropriate slices.
+ * Delegate the responsibility to applications and let them create cgroups
+   on their own, bypassing systemd completely.
+
+## Startup and shutdown best practices
+
+Question here are:
+
+ * Are there strong opionions on how the session-leader process should watch the users session units?
+ * Should systemd/logind/… provide an integrated way to define a session in terms of a running *user* unit?
+ * Is having `gnome-session-shutdown.target` that is run with `replace-irreversibly` considered a good practice?
+
+## Session Creation and Authentication
+
+Questions/thoughts here are:
+
+ * Who should be allowed to handle credentials? Not allowing this could increase security in particular with homed.
+ * It would make sense to use dynamic users for greeter sessions. Could we do something similar for normal sessions?
```

Comment:
This a whole another can of worms. Maybe let's treat this is as out of scope for now?

Reply:
Sure.